Riot/Model/Room/RoomBubbleCellData.m

```diff
@@ -104,12 +110,55 @@ - (NSAttributedString*)attributedTextMessage
     return attributedTextMessage;
 }
 
+#pragma mark - Bubble collapsing
+
+- (BOOL)collaspseWith:(id<MXKRoomBubbleCellDataStoring>)cellData
+{
+    if (self.tag == RoomBubbleCellDataTagMembership
+        && cellData.tag == RoomBubbleCellDataTagMembership)
+    {
+        // Keep a pagination between events of different days
+        NSString *bubbleDateString = [roomDataSource.eventFormatter dateStringFromDate:self.date withTime:NO];
+        NSString *eventDateString = [roomDataSource.eventFormatter dateStringFromDate:((RoomBubbleCellData*)cellData).date withTime:NO];
+        if (bubbleDateString && eventDateString && [bubbleDateString isEqualToString:eventDateString])
+        {
+            return YES;
+        }
+        else
+        {
+            return NO;
+        }
+    }
+    
+    return [super collaspseWith:cellData];
+}
+
+- (void)setCollapsed:(BOOL)collapsed
+{
+    if (collapsed != self.collapsed)
+    {
+        super.collapsed = collapsed;
+
+        // Refresh only cells serie header
+        if (self.collapsedAttributedTextMessage && self.nextCollapsableCellData)
```

Comment:
Why do you check "self.nextCollapsableCellData"?
I think self.nextCollapsableCellData could not be nil if self.collapsedAttributedTextMessage  is set.